Foundation/NSNotificationQueue.swift

```diff
@@ -15,26 +23,128 @@ public struct NSNotificationCoalescing : OptionSetType {
 }
 
 public class NSNotificationQueue : NSObject {
-    
+
+    private typealias NotificationQueueList = Array<NSNotificationQueue>
+    private typealias NSNotificationListEntry = (NSNotification, [String])
+    private typealias NSNotificationList = [NSNotificationListEntry]
+
+    private static let _notificationQueueThreadKey = "NSNotificationQueueThreadKey"
+    private static let _notificationListThreadKey = "NSNotificationListThreadKey"
+    private let notificationCenter: NSNotificationCenter
+    private var asapList = NSNotificationList()
+    private var idleList = NSNotificationList()
+    private var notificationQueueList: NotificationQueueList {
+        get {
+            let currentThread = NSThread.currentThread()
```

Comment:
you could probably use a TLS specific storage via this internal class `NSThreadSpecific< NotificationQueueList >()`
see `NSThread.currentThread()` for an example of how to use that.